### PR TITLE
[ML] `XGBoost`&`LightGBMTrainer` interfaces

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

Adds interfaces for `XGBoostTrainer` and `LightGBMTrainer`.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — COMMENTED** [review]
Nice! Left some minor comments for lightgbm- but same apply to xgboost.

File: python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        lightgbm_config: LightGBM training params. Refer to LightGBM documentation
+            for a list of possible parameters.
+        scaling_config: Configuration for how to scale data parallel training.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
+            the key "train" to denote which dataset is the training
+            dataset. If a ``preprocessor`` is provided and has not already been fit,
+            it will be fit on the training dataset. All datasets will be transformed
+            by the ``preprocessor`` if one is provided. All non-training
+            datasets will be used for validation.
```
[the remark this answers is not in the document]
Comment (amogkam) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py:47]: (can we clarify in the docstring)

[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py]: Done

File: python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                scaling_config={"num_workers": 3},
+                datasets={"train": train_dataset}
+            )
+            result = trainer.fit()
+
+
+    Args:
+        label_column: Name of the label column. A column with this name
+            must be present in the training dataset.
+        lightgbm_config: LightGBM training params. Refer to LightGBM documentation
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py]: Done

File: python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    Args:
+        label_column: Name of the label column. A column with this name
+            must be present in the training dataset.
+        lightgbm_config: LightGBM training parameters. Refer to
+            `LightGBM documentation <https://lightgbm.readthedocs.io/en\
+/latest/Parameters.html>`_ for a list of possible parameters.
+        scaling_config: Configuration for how to scale data parallel training.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Use
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py:43]: Apparently we aren't doing the splitting anymore.

File: python/ray/ml/train/integrations/xgboost/xgboost_trainer.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            a separate metric.
+        preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        label_column: str,
+        xgboost_config: Optional[Dict[str, Any]] = None,
```
Comment (maxpumperla) [inline python/ray/ml/train/integrations/xgboost/xgboost_trainer.py:58]: same as above.

File: python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            datasets will be used as separate validation sets, each reporting
+            a separate metric.
+        preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        label_column: str,
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py:57]: label is the only mandatory argument, every other is optional and following the same order as in other Trainers

File: python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        lightgbm_config: Optional[Dict[str, Any]] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
```
Comment (Yard1) [inline python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py:60]: question, should this be made a mandatory argument? the train dataset is always required.

Comment (amogkam) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py:60]: IMO, I think so! Like you said, train dataset is always required anyways.

Comment (Yard1) [reply python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py]: ok let's go with that

**Review by richardliaw — APPROVED** [review]

**Review by amogkam — APPROVED** [review]
